File: frontend/demo/component/badge/badge-number-only.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/badge';
import '@vaadin/horizontal-layout';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('badge-number-only')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-horizontal-layout theme="spacing">
        <!-- tag::snippet[] -->
        <vaadin-badge number="12" theme="number-only">New messages</vaadin-badge>
        <vaadin-badge number="3" theme="error number-only">Alerts</vaadin-badge>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/badge/BadgeNumberOnly.java
package com.vaadin.demo.component.badge;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.badge.Badge;
import com.vaadin.flow.component.badge.BadgeVariant;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;

@Route("badge-number-only")
public class BadgeNumberOnly extends HorizontalLayout {

    public BadgeNumberOnly() {
        // tag::snippet[]
        Badge inbox = new Badge("New messages", 12);
        inbox.addThemeVariants(BadgeVariant.NUMBER_ONLY);

        Badge alerts = new Badge("Alerts", 3);
        alerts.addThemeVariants(BadgeVariant.ERROR, BadgeVariant.NUMBER_ONLY);
        // end::snippet[]

        add(inbox, alerts);
    }

    public static class Exporter extends DemoExporter<BadgeNumberOnly> { // hidden-source-line
    } // hidden-source-line
}
